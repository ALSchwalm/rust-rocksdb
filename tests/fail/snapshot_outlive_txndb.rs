use rocksdb::TxnDB;

fn main() {
    let _snapshot = {
        let db = TxnDB::open_default("foo").unwrap();
        db.snapshot()
    };
}