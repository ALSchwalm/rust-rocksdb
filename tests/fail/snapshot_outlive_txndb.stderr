error[E0597]: `db` does not live long enough
 --> tests/fail/snapshot_outlive_txndb.rs:6:9
  |
4 |     let _snapshot = {
  |         --------- borrow later stored here
5 |         let db = TxnDB::open_default("foo").unwrap();
6 |         db.snapshot()
  |         ^^ borrowed value does not live long enough
7 |     };
  |     - `db` dropped here while still borrowed
